### DISCUSSION TITLE
[Question] Why Slack?

### DISCUSSION THREAD

**Discussion by uncomfyhalomacro** [unisonweb/unison#3268] [General]
Not trying to offend anyone here. Just wondering why not on other platforms or something like Element, Zulip, or even Discord. Anyway, interesting language, I may learn this with Haskell during my free time.

**Comment by pchiusano** [comment]
Hi there. It's been a while since we looked into this. A long time ago we were using Gitter, which had no threading and notifications didn't work very well. Zulip was considered, but Slack seemed like the nicest option and it was something most people were quite familiar with already.

We're not opposed to considering something else, though. :)

**Comment by zetashift** [comment]
I don't mind Slack that much, and indeed it's something a lot of persons are familiar with, but I think one reason to move away from slack is the 90 days retention period for messages. It slightly cripples the usefulness of `#beginner` and `#troubleshooting` channels.

Maybe it's worth holding a poll in the slack who wouldn't mind moving over to something more history-preserving-friendly?

As for what alternatives, I like Discord or Zulip!

**Comment by Nezteb** [comment]
I posted about this on the Unison Slack but I'll echo that here as well!

Slack is less-than-ideal for OSS communities because of the history limitation (among other things). If the core team is willing to consider alternatives, there's a few:

- [Matrix](https://matrix.org/) (with a client like [Element](https://element.io/))
- [IRC](https://libera.chat/)
- [Zulip](https://zulip.com/)
- [Linen](https://www.linen.dev/)
- [Discourse Chat](https://blog.discourse.org/2022/08/discourse-chat-is-now-available-try-it-out-today/)
- [Discord](https://discord.com/)

Another option (from @rlmark) is to keep the Slack but set up a mirror to another one of the above options; that would give people a way to see a full chat history too!

**Reply by bytesource** [reply]
My only experience is with F# Slack. While many people like it, the lack of available older messages severely limits the search function. This leads to the same questions getting asked again and again.  

I suggest choosing a platform that is indexed by Google to get the word out. Programming questions are typically asked on Google, and users assume that documentation or adaptation is lacking if there are no results.

The Elixir forum, https://elixirforum.com/, is a good example, where most of my Elixir-related questions in Google return one or more forum entries.

**Reply by uncomfyhalomacro** [reply]
I pretty much like the idea of using Matrix and Discourse, where Matrix is for chat and Discourse is for long posts and discussions

**Reply by zetashift** [reply]
IMO mentioned above https://www.linen.dev/ looks like a great middleground

**Reply by Nezteb** [reply]
Apparently a few of these already exist:

- Matrix: https://matrix.to/#/#unison:matrix.org
- Unison:
  - https://www.linen.dev/invite/unison-lang (Someone else by the name of `r.hackr` made this)
  - https://www.linen.dev/invite/unison (I created this one before I realized the one above existed)

@pchiusano @runarorama @rlmark What do y'all think?

**Comment by ceedubs** [comment]
A while back communication moved from Slack to the [Unison Discord](https://unison-lang.org/discord).
